src/bson/bson-config.h

```diff
@@ -57,26 +57,159 @@
 #endif
 
 
+
 /*
- * Define to 1 if we have access to GCC 32-bit atomic builtins.
- * While this requires GCC 4.1+ in most cases, it is also architecture
- * dependent. For example, some PPC or ARM systems may not have it even
- * if it is a recent GCC version.
+ * Define to 1 if you have clock_gettime() available.
  */
-#define BSON_HAVE_ATOMIC_32_ADD_AND_FETCH 0
-#if BSON_HAVE_ATOMIC_32_ADD_AND_FETCH != 1
-# undef BSON_HAVE_ATOMIC_32_ADD_AND_FETCH
+#ifdef HAVE_CLOCK_GETTIME
+# define BSON_HAVE_CLOCK_GETTIME 1
+#endif
+
+#if BSON_HAVE_CLOCK_GETTIME != 1
+# undef BSON_HAVE_CLOCK_GETTIME
 #endif
 
+
 /*
- * Similarly, define to 1 if we have access to GCC 64-bit atomic builtins.
+ * Define to 1 if you have strings.h available on your platform.
  */
-#define BSON_HAVE_ATOMIC_64_ADD_AND_FETCH 0
-#if BSON_HAVE_ATOMIC_64_ADD_AND_FETCH != 1
-# undef BSON_HAVE_ATOMIC_64_ADD_AND_FETCH
+#define BSON_HAVE_STRINGS_H 0
+#if BSON_HAVE_STRINGS_H != 1
+# undef BSON_HAVE_STRINGS_H
 #endif
 
 
+/*
+ * Define to 1 if you have strnlen available on your platform.
+ */
+#define BSON_HAVE_STRNLEN 0
+#if BSON_HAVE_STRNLEN != 1
+# undef BSON_HAVE_STRNLEN
+#endif
+
+
+/*
+ * Define to 1 if you have snprintf available on your platform.
+ */
+#ifdef MS_WINDOWS
+# define BSON_HAVE_SNPRINTF 0
+#else
+# define BSON_HAVE_SNPRINTF 1
+#endif
+
+#if BSON_HAVE_SNPRINTF != 1
+# undef BSON_HAVE_SNPRINTF
+#endif
+
+
+/*
+ * Define to 1 if you have gmtime_r available on your platform.
+ */
+#ifdef MS_WINDOWS
+# define BSON_HAVE_GMTIME_R 0
+#else
+# define BSON_HAVE_GMTIME_R 1
+#endif
+
+#if BSON_HAVE_GMTIME_R != 1
+# undef BSON_HAVE_GMTIME_R
+#endif
+
+
+
+/*
+ * Define to 1 if you have struct timespec available on your platform.
+ */
+#ifdef HAVE_CLOCK_GETTIME
+# define BSON_HAVE_TIMESPEC 1
+#endif
+
+#if BSON_HAVE_TIMESPEC != 1
+# undef BSON_HAVE_TIMESPEC
+#endif
+
+
+/*
+ * Define to 1 if you want extra aligned types in libbson
+ */
+#define BSON_EXTRA_ALIGN 1
+#if BSON_EXTRA_ALIGN != 1
+# undef BSON_EXTRA_ALIGN
+#endif
```

Comment:
BSON_EXTRA_ALIGN and other defines already exist in this file.

Reply:
Done.